fix(Library): Fixed playlist name incorrectly used as path for importing

DidierMalenfant opened the pull request:
As mentioned in #15645, library importers build TreeItems with the name of the playlist as the label and the path to the playlist as data. Therefore createPlaylistModelForPlaylist() should get the data member of the TreeItem and not the label in order for the playlist to be found during import.

DidierMalenfant committed:
Library: Fixed playlist name incorrectly used as path for importing

Importers build TreeItems with the name of the playlist as the label and the path to the playlist as data. Therefore createPlaylistModelForPlaylist() should get the data member of the TreeItem and not the label in order for the playlist to be found.

DidierMalenfant committed:
Library: Line length fixes

Per coding standard

DidierMalenfant committed:
Library: Made a local variable explicitly const

ronso0 commented on src/library/baseexternallibraryfeature.cpp:
```diff
@@ -165,22 +166,29 @@ void BaseExternalLibraryFeature::appendTrackIdsFromRightClickIndex(
-            createPlaylistModelForPlaylist(*pPlaylist));
+    const auto* pTreeItem = static_cast<TreeItem*>(
+            m_lastRightClickedIndex.internalPointer());
+    *pPlaylist = pTreeItem->getLabel();
```
```suggestion
    const auto* pTreeItem = static_cast<TreeItem*>(
            m_lastRightClickedIndex.internalPointer());
    VERIFY_OR_DEBUG_ASSERT(pTreeItem) {
        return;
    }
    *pPlaylist = pTreeItem->getLabel();
```

DidierMalenfant committed:
chore(ExternalLibraryFeature): Fixed typos in a comment

DidierMalenfant committed:
fix(ExternalLibraryFeature): Added a debug assert

DidierMalenfant committed:
fix(ExternalLibraryFeature): Refactored createPlaylistModelForPlaylist()

This now receives a QVariant which is the data part of the TreeItem the request to create the playlist originates from. This allow each implementation of the LibraryFeature class to correctly retrieve the playlist path according to how it stored it in there in the first place.

This fixes all playlist importing from other programs like Rekordbox, Serato, etc.. which were silently failing before.

DidierMalenfant committed:
Merge branch '2.5' into playlist_import_fixes

daschuer commented:
Thank you for all the work. Today is the planned release date for 2.5.4. Let's see if this can slip in.

daschuer commented on src/library/rekordbox/rekordboxfeature.cpp:
```diff
@@ -1411,10 +1411,10 @@ void RekordboxFeature::htmlLinkClicked(const QUrl& link) {
     auto pModel = std::make_unique<RekordboxPlaylistModel>(
             this, m_pLibrary->trackCollectionManager(), m_trackSource);
-    pModel->setPlaylist(playlist);
+    pModel->setPlaylist(data.toList()[0].toString());
```
```suggestion
    if (data.canConvert<QVariantList>()) {
        QVariantList playlists = data.toList();
        if (playlists.size() > 0) {
            pModel->setPlaylist(playlists.at(0).toString());
        }
    }
```

daschuer commented on src/library/itunes/itunesfeature.cpp:
```diff
@@ -129,14 +129,14 @@ ITunesFeature::~ITunesFeature() {
             "itunes_playlist_tracks",
             m_trackSource);
-    pModel->setPlaylist(playlist);
+    pModel->setPlaylistById(data.toInt());
```
```suggestion
    bool ok; 
    int playlistId = data.toInt(&ok);
    if (ok) {
        pModel->setPlaylistById(playlistId);
    }
```

daschuer reviewed (commented):
BaseExternalPlaylistModel::setPlaylist() can make use of a 
```
    VERIFY_OR_DEBUG_ASSERT(!playlist_path.isEmpty()) {
        return;
    }
``

Swiftb0y reviewed (commented):
yeah, `DEBUG_ASSERT` the QVariant conversions, LGTM otherwise
